miner/worker.go

```diff
@@ -1055,6 +1059,10 @@ loop:
 			}
 			return atomic.LoadInt32(interrupt) == commitInterruptNewHead, circuitCapacityReached
 		}
+		// seal block early early if we're over time
+		if w.current.tcount > 0 && w.chainConfig.Clique != nil && uint64(time.Now().Unix()) > w.current.parentTimestamp+w.chainConfig.Clique.Period {
```

Comment:
You mean because we set `header.Time` it here?

https://github.com/scroll-tech/go-ethereum/blob/d81a3308b18232c134335a6cfe59cf3df0611658/consensus/clique/clique.go#L557-L560

I guess I was worried about the case where `parentTimestamp + clique.Period < time.Now()` and so `current.header.Timestamp == time.Now()`. But it's probably not an issue, if line 559 is triggered we're already over time and whenever `time.Now() > current.header.Timestamp` we can seal the block.

I'll update.

Reply:
Updated in [29a3cb9](https://github.com/scroll-tech/go-ethereum/pull/706/commits/29a3cb925481e07d75832641bad80aa7e1b933a4).

Reply:
LGTM will approve once it's updated